docker/debug/Dockerfile

```diff
@@ -11,26 +10,35 @@ RUN cat rust-toolchain | xargs rustup toolchain install
 
 # Install protoc, openssl, and pkg-config
 RUN apt-get update && \
+    apt-get install -y build-essential && \
+    apt-get install -y wget && \
+    apt-get install -y perl && \
+    apt-get install -y gcc && \
     apt-get install -y pkg-config && \
     apt-get install -y protobuf-compiler && \
-    apt-get install -y libssl-dev
-
-COPY Cargo.toml Cargo.lock ./
-COPY ./core/Cargo.toml ./core/Cargo.toml
-COPY ./circuits ./circuits
-COPY ./crypto ./crypto
-COPY ./circuit-macros ./circuit-macros
-COPY ./integration-helpers ./integration-helpers
+    apt-get install -y libclang-dev && \
+    apt-get install -y libssl-dev && \
+    apt-get install -y libz-dev
+
+# Install libssl3 from source
+RUN wget https://www.openssl.org/source/openssl-3.0.10.tar.gz
+RUN tar -xzvf openssl-*.tar.gz && rm -rf openssl-*.tar.gz
+WORKDIR /build/openssl-3.0.10
+RUN ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl shared zlib
+RUN make && make install
+WORKDIR /build
 
-# Create a dummy entrypoint to build the dependencies
-RUN mkdir core/src
-RUN echo 'fn main() { println!("dummy main!") }' >> core/src/main.rs
+# Copy the entire workspace and remove all source files, except for dummy lib entrypoints
```

Comment:
nit: This comment needs more detail, the shell commands are hard to read, maybe something like:
```
# Copy the workspace, remove all files except `lib.rs`, and add clear the file contents of each `lib`